verification/tagTCI.go

```diff
@@ -3,55 +3,239 @@
+	}
+
+	// Get parent context TCI values for cumulative value calculation
+	parentTci, err := c.GetTaggedTCI(defaultCtxTCITag)
+	if err != nil {
+		t.Fatalf("Could not get tagged TCI: %v", err)
+	}
+
+	// Cross-verify parent's TCI_CUMULATIVE
+	hasher.Write(defaultTci)
+	hasher.Write(parentTci.CurrentTCI)
+	wantCumulativeTCI = hasher.Sum(nil)
+	if !bytes.Equal(parentTci.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("Parent node's cumulative TCI %x, expected %x", parentTci.CumulativeTCI, wantCumulativeTCI)
+	}
+
+	// Preserve parent context to restore for subsequent tests.
+	parentHandle, err := c.RotateContextHandle(handle, RotateContextHandleFlags(0))
+	if err != nil {
+		t.Errorf("[ERROR]: Error while rotating parent context handle, this may cause failure in subsequent tests: %s", err)
+	}
+
+	// Derive Child context with input data, tag it and check TCI_CUMULATIVE
+	childCtx, err := c.DeriveChild(parentHandle, tciValue, DeriveChildFlags(RetainParent), 0, 0)
+	if err != nil {
+		t.Fatalf("[FATAL]: Error while creating default child handle in default context: %s", err)
+	}
+
+	// Tag derived context
+	newHandle, err := c.TagTCI(&childCtx.NewContextHandle, childCtxTCITag)
+	if err != nil {
+		t.Fatalf("Could not tag TCI: %v", err)
 	}
 
-	// TODO: When DeriveChild is implemented, call it here to add more TCIs and call TagTCI again.
+	childTci, err := c.GetTaggedTCI(childCtxTCITag)
+	if err != nil {
+		t.Fatalf("Could not get tagged TCI: %v", err)
+	}
+
+	if !bytes.Equal(childTci.CurrentTCI, tciValue) {
+		t.Errorf("GetTaggedTCI returned current TCI %x, expected %x", childTci.CurrentTCI, tciValue)
+	}
+
+	// Check TCI_CUMULATIVE after creating child context
+	hasher.Reset()
+	hasher.Write(defaultTci)
+	hasher.Write(childTci.CurrentTCI)
+	wantCumulativeTCI = hasher.Sum(nil)
+	if !bytes.Equal(childTci.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("Child node's cumulative TCI %x, expected %x", childTci.CumulativeTCI, wantCumulativeTCI)
+	}
+
+	// Extend TCI to child context and check TCI_CURRENT and TCI_CUMULATIVE
+	newHandle, err = c.ExtendTCI(newHandle, extendTciValue)
+	if err != nil {
+		t.Fatalf("Could not tag TCI: %v", err)
+	}
+
+	childExtendTci, err := c.GetTaggedTCI(childCtxTCITag)
+	if err != nil {
+		t.Fatalf("Could not get tagged TCI: %v", err)
+	}
+
+	if !bytes.Equal(childExtendTci.CurrentTCI, extendTciValue) {
+		t.Errorf("GetTaggedTCI returned current TCI %x, expected %x", childExtendTci.CurrentTCI, extendTciValue)
+	}
+
+	// Check TCI_CUMULATIVE after extending input to child context
+	hasher.Reset()
+	hasher.Write(childTci.CumulativeTCI)
+	hasher.Write(childExtendTci.CurrentTCI)
+	wantCumulativeTCI = hasher.Sum(nil)
+	if !bytes.Equal(childExtendTci.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("Child node's cumulative TCI %x, expected %x", childExtendTci.CumulativeTCI, wantCumulativeTCI)
+	}
+
+	// Clean up derived context and restore default context handle for subsequnt tests
```

Comment:
Code to cleanup derived context so that other tests iin the line are not affected and report unexpected errors.